includes/blocks/class-sensei-course-list-student-course-filter.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * File containing the Sensei_Course_List_Student_Course_Filter class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_List_Student_Course_Filter
+ */
+class Sensei_Course_List_Student_Course_Filter extends Sensei_Course_List_Filter_Abstract {
+
+	/**
+	 * Name of the filter.
+	 */
+	const FILTER_NAME = 'student_course';
+
+	/**
+	 * Unique key for the filter param.
+	 *
+	 * @var string
+	 */
+	private $param_key = 'course-list-student-course-filter-';
+
+	/**
+	 * Options for filter dropdown.
+	 *
+	 * @var array
+	 */
+	private $filter_options = [];
+
+	/**
+	 * Constructor for Sensei_Course_List_Featured_Filter class.
+	 */
+	public function __construct() {
+		$this->filter_options = [
+			'all'       => __( 'All Courses', 'sensei-lms' ),
+			'active'    => __( 'Active', 'sensei-lms' ),
+			'completed' => __( 'Completed', 'sensei-lms' ),
+		];
+	}
+	/**
+	 * Get the content to be be rendered inside the filtered block.
+	 *
+	 * @param int $query_id The id of the Query block this filter is rendering inside.
+	 */
+	public function get_content( $query_id ) : string {
+		if ( empty( get_current_user_id() ) ) {
+			return '';
+		}
+		$selected_option  = 'all';
+		$filter_param_key = $this->param_key . $query_id;
+		// phpcs:ignore WordPress.Security.NonceVerification
+		if ( isset( $_GET[ $filter_param_key ] ) ) {
+			$selected_option = sanitize_text_field( wp_unslash( $_GET[ $filter_param_key ] ) ); // phpcs:ignore WordPress.Security.NonceVerification
```

Comment:
Maybe we should put it into a variable to avoid having to add the PHPCS rule twice.

Reply:
Thanks! Taken a slightly different approach [here](https://github.com/Automattic/sensei/pull/5578/commits/26fb4149f78a048bfc2bd76add22b9b0f270f45f), Thanks!